packages/store/addon/-private/system/snapshot.ts

```diff
@@ -98,34 +143,33 @@ export default class Snapshot {
    @property record
    @type {Model}
    */
-  get record() {
+  get record(): RecordInstance {
     return this._internalModel.getRecord();
   }
 
-  get _attributes() {
-    let attributes = this.__attributes;
-
-    if (attributes === null) {
-      let record = this.record;
-      attributes = this.__attributes = Object.create(null);
-      let attrs;
+  get _attributes(): Dict<any> {
+    if (this.__attributes !== null) {
+      return this.__attributes;
+    }
+    let record = this.record;
+    const attributes = (this.__attributes = Object.create(null));
+    let attrs: string[];
 
-      if (CUSTOM_MODEL_CLASS) {
-        attrs = Object.keys(this._store._attributesDefinitionFor(this.modelName, this.identifier));
-      } else {
-        attrs = Object.keys(this._store._attributesDefinitionFor(this.modelName, record.id));
-      }
-      if (CUSTOM_MODEL_CLASS) {
-        attrs.forEach(keyName => {
-          if (this.type.isModel) {
-            attributes[keyName] = get(record, keyName);
-          } else {
-            attributes[keyName] = recordDataFor(this._internalModel).getAttr(keyName);
-          }
-        });
-      } else {
-        record.eachAttribute(keyName => (attributes[keyName] = get(record, keyName)));
-      }
+    if (CUSTOM_MODEL_CLASS) {
+      attrs = Object.keys(this._store._attributesDefinitionFor(this.modelName, this.identifier));
+    } else {
+      attrs = Object.keys(this._store._attributesDefinitionFor(this.modelName));
+    }
```

Comment:
Non-TS, and mostly a point of curiosity (esp. since it predates this) – any particular preference for `let` and the `if`/`else` vs. just doing this?

```ts
const attrs = CUSTOM_MODEL_CLASS
  ? Object.keys(this._store._attributesDefinitionFor(this.modelName, this.identifier))
  : Object.keys(this._store._attributesDefinitionFor(this.modelName));
```

Reply:
no :) I prefer using a ternary here but I think we reach for the if/else out of habit.

Reply:
@chriskrycho I think for feature flags specifically, if/else are easier to scan & read because you can just jump between blocks, and possibly collapse them compared to interpreting the ternary